.NET/Microsoft.Recognizers.Text.Number/English/Extractors/NumberExtractor.cs

```diff
@@ -10,29 +10,33 @@ public class NumberExtractor : BaseNumberExtractor
     {
         internal sealed override ImmutableDictionary<Regex, string> Regexes { get; }
 
+        protected sealed override NumberOptions Options { get; }
+
         protected sealed override string ExtractType { get; } = Constants.SYS_NUM; // "Number";
 
         protected sealed override Regex NegativeNumberTermsRegex { get; }
 
-        private static readonly ConcurrentDictionary<string, NumberExtractor> Instances = new ConcurrentDictionary<string, NumberExtractor>();
-
-        public static NumberExtractor GetInstance(NumberMode mode = NumberMode.Default) {
-
-            var placeholder = mode.ToString();
+        private static readonly ConcurrentDictionary<(NumberMode, NumberOptions), NumberExtractor> Instances =
+            new ConcurrentDictionary<(NumberMode, NumberOptions), NumberExtractor>();
 
-            if (!Instances.ContainsKey(placeholder))
+        public static NumberExtractor GetInstance(NumberMode mode = NumberMode.Default, NumberOptions options = NumberOptions.None)
```

Comment:
Should we define a 'config', like in DateTime?

Reply:
For now, we don't have many things to configure, so I think we can add it later.